File: data/data_generator.py
#/usr/bin/env python3

import pandas as pd
import numpy as np

no_otus = 100
no_samples = 50

sample_names =["sample" + str(x) for x in range(1,no_samples + 1)]  
row_names = ["org" + str(x) for x in range(1,no_otus + 1)] 
col_names = row_names

# OTU table
otu = pd.DataFrame(np.random.randint(0,10000,size = (no_otus, no_samples)), index = row_names,  columns = sample_names)
otu.to_csv('otu_table.tsv', header=True, index=True, sep = "\t")

# correlation values
corr_val = pd.DataFrame(np.random.uniform(-1,1, size = (no_otus, no_otus)), index = row_names, columns = col_names)
corr_val.to_csv('random_correlation.tsv', header=True, index=True, sep = "\t")

# p values
p_val = pd.DataFrame(np.random.uniform(0,0.08, size = (no_otus, no_otus)), index = row_names,  columns = col_names)
p_val.to_csv('random_p_value.tsv', header=True, index=True, sep = "\t")

File: data/data_generator.r
org <- 10
samples <- 50

# otu table
otu <- matrix(sample.int(500, org * samples, replace = TRUE), org, samples)
rownames(otu) <- paste0("org", seq(1, org, 1))
colnames(otu) <- paste0("sample", seq(1, samples, 1))
write.table(otu, "otu_table.tsv", quote = FALSE, sep = "\t", col.names = TRUE)

# p value
p_val <- matrix(runif(org, min = 0, max = 0.07), org, org)
rownames(p_val) <- paste0("org", seq(1, org, 1))
colnames(p_val) <- paste0("org", seq(1, org, 1))
write.table(p_val, "random_p_value.tsv", quote = FALSE, sep = "\t", col.names = TRUE)

# corr value
corr_val <- matrix(runif(org, min = -1, max = 1), org, org)
rownames(corr_val) <- paste0("org", seq(1, org, 1))
colnames(corr_val) <- paste0("org", seq(1, org, 1))
write.table(corr_val, "random_correlation.tsv", quote = FALSE, sep = "\t", col.names = TRUE)